bdns/dns_test.go

```diff
@@ -368,7 +368,8 @@ func TestDNSLookupHost(t *testing.T) {
 	t.Logf("%s - IP: %s, Err: %s", hostname, ip, err)
 	test.AssertError(t, err, "Should be an error")
 	expectedErr := DNSError{dns.TypeA, hostname, nil, dns.RcodeRefused}
-	if err, ok := err.(*DNSError); !ok || *err != expectedErr {
+	var dnserr *DNSError
+	if !errors.As(err, &dnserr) || *dnserr != expectedErr {
```

Comment:
When you have !a || !b you can equally well transform that into !(a && b) due to [DeMorgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws). (You may already know this but being explicit just in case).

So, which style you choose in any given situation is largely a matter of style and clarity. One rule of thumb I just made up on the spot is to always try to minimize the number of "nots" (!'s) in an expression. Another way to make the call in this particular situation: We expect to see a lot of `if errors.As(err, &foo) && foo.Bar()`, so wrapping that in a not makes sense and follows idioms people already expect to see: `if !(errors.As(err, &foo) && foo.Bar())`, or in this case:

```suggestion
	if !(errors.As(err, &dnserr) && *dnserr == expectedErr) {
```

This is particularly useful because errors.As requires you to reason about how dnserr is only valid when errors.As returns true.